internal/execute/execute.go

```diff
@@ -0,0 +1,120 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package execute
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/submariner-operator/api/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/internal/constants"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/client"
+	operatorClientset "github.com/submariner-io/submariner-operator/pkg/client/clientset/versioned"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	subClientsetv1 "github.com/submariner-io/submariner/pkg/client/clientset/versioned"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Cluster struct {
+	Config     *rest.Config
+	Name       string
+	KubeClient kubernetes.Interface
+	DynClient  dynamic.Interface
+	SubmClient subClientsetv1.Interface
+	Submariner *v1alpha1.Submariner
```

Comment:
I think we should provide a method to lazily retrieve this so third party users don't have to do it. 

```suggestion
	submarinerOrErr *interface{}
	
	...
	func (c *Cluster) GetSubmariner() (*v1alpha1.Submarine, error) {
	    if c.submarinerOrErr == nil {
		c.submarinerOrErr = new(interface{})
		
		submariner, err := c.ClientProducer.ForOperator().SubmarinerV1alpha1().Submariners(constants.SubmarinerNamespace).
		Get(context.TODO(), constants.SubmarinerName, metav1.GetOptions{})
		if !apierrors.IsNotFound(err) {
			if err != nil {
				*c.submarinerOrErr = errors.Wrap(err, "Error retrieving Submariner resource")
			} else {
				*c.submarinerOrErr = submariner
			}
		    }
	        }

	        if *c.submarinerOrErr == nil {
		    return nil, nil
	        }

                if err, ok := (*c.submarinerOrErr).(error); ok {
		    return nil, err
	        }

	        return (*c.submarinerOrErr).(*Submariner), nil
	   }
```

Reply:
Ditto